components/x-gift-article/src/lib/api.js

```diff
@@ -43,9 +43,16 @@ export default class ApiClient {
 		}
 	}
 
-	async getGiftUrl(articleId) {
+	async getGiftUrl(articleId, highlightsToken = undefined) {
 		try {
-			const json = await this.fetchJson('/article/gift-link/' + encodeURIComponent(articleId))
+			let json
+			if (highlightsToken) {
+				json = await this.fetchJson(
+					'/article/gift-link/' + encodeURIComponent(articleId) + `?highlightsToken=${highlightsToken}`
+				)
+			} else {
+				json = await this.fetchJson('/article/gift-link/' + encodeURIComponent(articleId))
```

Comment:
**Note:** Along with the above comment on like 51 lets also make this consistent with whatever we choose for that comment